includes/admin/meta-boxes/class-wc-meta-box-order-data.php

```diff
@@ -482,49 +485,28 @@ public static function save( $post_id, $post ) {
 				$payment_method_title = $methods[ $payment_method ]->get_title();
 			}
 
-			update_post_meta( $post_id, '_payment_method', $payment_method );
-			update_post_meta( $post_id, '_payment_method_title', $payment_method_title );
+			$props['payment_method'] = $payment_method;
+			$props['payment_method_title'] = $payment_method_title;
 		}
 
-		// Update date
+		// Update date.
 		if ( empty( $_POST['order_date'] ) ) {
 			$date = current_time( 'timestamp' );
 		} else {
 			$date = strtotime( $_POST['order_date'] . ' ' . (int) $_POST['order_date_hour'] . ':' . (int) $_POST['order_date_minute'] . ':00' );
 		}
 
-		$date = date_i18n( 'Y-m-d H:i:s', $date );
-
-		$wpdb->query( $wpdb->prepare( "UPDATE $wpdb->posts SET post_date = %s, post_date_gmt = %s WHERE ID = %s", $date, get_gmt_from_date( $date ), $post_id ) );
-
-		clean_post_cache( $post_id );
-
-		// If customer changed, update any downloadable permissions
-		if ( $customer_changed ) {
-			$wpdb->update( $wpdb->prefix . "woocommerce_downloadable_product_permissions",
-				array(
-					'user_id'    => absint( get_post_meta( $post->ID, '_customer_user', true ) ),
-					'user_email' => wc_clean( get_post_meta( $post->ID, '_billing_email', true ) ),
-				),
-				array(
-					'order_id' 		=> $post_id,
-				),
-				array(
-					'%d',
-					'%s',
-				),
-				array(
-					'%d',
-				)
-			);
-		}
+		$props['date_created'] = $date;
+
+		clean_post_cache( $order_id );
```

Comment:
If this is needed, it should move to the CRUD class save method if not there already.